import pandas as pd
df_abmd = pd.read_csv('/Users/dongyuxin/Downloads/justynaCase/ABMD.csv')
df_abbv= pd.read_csv('/Users/dongyuxin/Downloads/justynaCase/ABBV.csv')
df_abc= pd.read_csv('/Users/dongyuxin/Downloads/justynaCase/ABC.csv')
df_aapl= pd.read_csv('/Users/dongyuxin/Downloads/justynaCase/AAPL.csv')

In [6]:
df_abmd.head()

Date        High         Low        Open       Close  Volume  \
0  2017-11-13  196.000000  193.360001  193.360001  195.130005  464600   
1  2017-11-14  196.080002  193.110001  195.000000  193.750000  343900   
2  2017-11-15  194.169998  190.070007  193.610001  193.110001  342000   
3  2017-11-16  196.479996  193.360001  193.789993  194.910004  291600   
4  2017-11-17  194.880005  191.440002  194.559998  194.320007  448800   

    Adj Close  
0  195.130005  
1  193.750000  
2  193.110001  
3  194.910004  
4  194.320007

In [9]:
df_abmd['Close'] == df_abmd['Adj Close']

0       True
1       True
2       True
3       True
4       True
        ... 
1254    True
1255    True
1256    True
1257    True
1258    True
Length: 1259, dtype: bool

In [10]:
df_abbv['Close'] == df_abbv['Adj Close']

0       False
1       False
2       False
3       False
4       False
        ...  
1254     True
1255     True
1256     True
1257     True
1258     True
Length: 1259, dtype: bool

In [11]:
df_abc['Close'] == df_abc['Adj Close']

0       False
1       False
2       False
3       False
4       False
        ...  
1254    False
1255    False
1256    False
1257     True
1258     True
Length: 1259, dtype: bool

In [12]:
df_aapl['Close'] == df_aapl['Adj Close']

0       False
1       False
2       False
3       False
4       False
        ...  
1254     True
1255     True
1256     True
1257     True
1258     True
Length: 1259, dtype: bool

Answer 1:There is no difference between Close and Adj Close column in company ABMD

In [26]:
df_abmd.sort_values('Adj Close',ascending = True)


Date        High         Low        Open       Close   Volume  \
590  2020-03-20  143.869995  130.009995  142.130005  130.509995   848800   
591  2020-03-23  134.740005  119.010002  132.389999  132.339996  1296500   
584  2020-03-12  149.429993  135.770004  143.070007  136.970001  1060800   
589  2020-03-19  145.949997  135.460007  142.410004  140.509995   950600   
600  2020-04-03  145.699997  140.820007  142.380005  141.399994   440800   
..          ...         ...         ...         ...         ...      ...   
147  2018-06-15  445.450012  435.480011  441.000000  443.579987   799000   
221  2018-10-01  459.750000  442.929993  452.899994  444.250000   526300   
218  2018-09-26  454.000000  438.519989  443.489990  444.700012   914000   
150  2018-06-20  450.929993  441.100006  441.100006  446.209991   583000   
220  2018-09-28  455.399994  441.000000  441.290009  449.750000   696400   

      Adj Close  
590  130.509995  
591  132.339996  
584  136.970001  
589  140.509995  
600  141.399994  
..          ...  
147  443.579987  
221  444.250000  
218  444.700012  
150  446.209991  
220  449.750000  

[1259 rows x 7 columns]

In [20]:
df_abbv.sort_values('Adj Close',ascending = True)

Date        High         Low        Open       Close      Volume  \
440   2019-08-15   64.239998   62.660000   64.120003   62.980000   9195700.0   
439   2019-08-14   64.989998   63.500000   64.500000   63.500000   9373300.0   
441   2019-08-16   64.809998   63.070000   63.509998   64.430000   8348100.0   
434   2019-08-07   64.900002   63.310001   64.379997   64.459999  10532100.0   
437   2019-08-12   65.480003   64.080002   65.239998   64.529999   5258900.0   
...          ...         ...         ...         ...         ...         ...   
1110  2022-04-12  170.559998  166.210007  169.309998  167.309998   8279700.0   
1106  2022-04-06  169.270004  163.800003  164.490005  168.910004   8810800.0   
1109  2022-04-11  175.039993  169.320007  174.899994  169.830002   8163200.0   
1107  2022-04-07  174.589996  167.539993  167.669998  173.279999   7469000.0   
1108  2022-04-08  175.910004  172.500000  173.000000  174.960007   7204200.0   

       Adj Close  
440    54.021832  
439    54.467869  
441    55.265583  
434    55.291313  
437    55.351360  
...          ...  
1110  162.706985  
1106  164.262970  
1109  165.157654  
1107  168.512726  
1108  170.146515  

[1259 rows x 7 columns]

In [21]:
df_abc.sort_values('Adj Close',ascending = True)

Date        High         Low        Open       Close     Volume  \
279   2018-12-24   72.709999   70.629997   71.970001   70.760002   523900.0   
359   2019-04-22   72.970001   70.550003   72.430000   71.059998  1270400.0   
278   2018-12-21   75.559998   71.550003   73.209999   71.980003  2573600.0   
357   2019-04-17   75.790001   71.629997   75.589996   72.300003  1587300.0   
358   2019-04-18   72.779999   71.279999   72.300003   72.449997  1431900.0   
...          ...         ...         ...         ...         ...        ...   
1111  2022-04-13  164.149994  161.369995  162.630005  162.960007  1075400.0   
1254  2022-11-07  163.610001  160.550003  161.940002  162.009995  1393600.0   
1116  2022-04-21  167.190002  164.110001  165.690002  165.020004   928800.0   
1108  2022-04-08  166.899994  163.039993  163.520004  165.440002  2019600.0   
1115  2022-04-20  166.470001  163.649994  163.649994  165.729996   924300.0   

       Adj Close  
279    66.344460  
359    66.935455  
278    67.488350  
357    68.103493  
358    68.244766  
...          ...  
1111  161.464783  
1254  161.513275  
1116  163.505875  
1108  163.922028  
1115  164.209366  

[1259 rows x 7 columns]

In [23]:
df_aapl.sort_values('Adj Close',ascending = True)

Date        High         Low        Open       Close       Volume  \
285   2019-01-03   36.430000   35.500000   35.994999   35.547501  365248800.0   
279   2018-12-24   37.887501   36.647499   37.037498   36.707500  148676800.0   
287   2019-01-07   37.207500   36.474998   37.174999   36.982498  219111200.0   
286   2019-01-04   37.137501   35.950001   36.132500   37.064999  234428400.0   
292   2019-01-14   37.817501   37.305000   37.712502   37.500000  129756800.0   
...          ...         ...         ...         ...         ...          ...   
1038  2021-12-29  180.630005  178.139999  179.330002  179.380005   62348900.0   
1026  2021-12-10  179.630005  174.690002  175.210007  179.449997  115402700.0   
1042  2022-01-04  182.940002  179.119995  182.630005  179.699997   99310400.0   
1036  2021-12-27  180.419998  177.070007  177.089996  180.330002   74919600.0   
1041  2022-01-03  182.880005  177.710007  177.830002  182.009995  104487900.0   

       Adj Close  
285    34.309589  
279    35.429188  
287    35.694607  
286    35.774235  
292    36.194088  
...          ...  
1038  178.344925  
1026  178.414520  
1042  178.663071  
1036  179.289444  
1041  180.959732  

[1259 rows x 7 columns]

Answer2: ABMD:highest and lowest price (Adj Close):449.750000; 130.509995
         ABBV:highest and lowest price (Adj Close):170.146515; 54.021832
         ABC:highest and lowest price (Adj Close):164.209366; 66.344460
         AAPL:highest and lowest price (Adj Close):180.959732; 34.309589
        

In [33]:
import numpy as np
logarithmic_returns1 = np.log(df_abmd['Adj Close'])
logarithmic_returns1.describe()

count    1259.000000
mean        5.618517
std         0.253474
min         4.871450
25%         5.521181
50%         5.660248
75%         5.789731
max         6.108692
Name: Adj Close, dtype: float64

In [34]:
import numpy as np
logarithmic_returns1 = np.log(df_abbv['Adj Close'])
logarithmic_returns1.describe()

count    1259.000000
mean        4.501442
std         0.276713
min         3.989388
25%         4.306940
50%         4.422301
75%         4.689019
max         5.136660
Name: Adj Close, dtype: float64

In [35]:
import numpy as np
logarithmic_returns1 = np.log(df_abc['Adj Close'])
logarithmic_returns1.describe()

count    1259.000000
mean        4.586290
std         0.232838
min         4.194860
25%         4.400476
50%         4.519552
75%         4.768891
max         5.101142
Name: Adj Close, dtype: float64

In [36]:
import numpy as np
logarithmic_returns1 = np.log(df_aapl['Adj Close'])
logarithmic_returns1.describe()

count    1259.000000
mean        4.408633
std         0.534165
min         3.535425
25%         3.877338
50%         4.364802
75%         4.944377
max         5.198275
Name: Adj Close, dtype: float64

Ansewer3: the logarithmic returns from Adj Close for each company reports are above

In [48]:
return_abmd = df_abmd['Adj Close']- df_abmd['Open']
return_abmd.sum()

15.519958496093523

In [49]:
return_abc = df_abc['Adj Close']- df_abc['Open']
return_abc.sum()

-4493.753379821777

In [50]:
return_aapl = df_aapl['Adj Close']- df_aapl['Open']
return_aapl.sum()

-1582.6297607421866

In [51]:
return_abbv = df_abbv['Adj Close']- df_abmd['Open']
return_abbv.sum()

-239403.7234954834

Answer6:abmd

TypeError: Only valid with DatetimeIndex, TimedeltaIndex or PeriodIndex, but got an instance of 'RangeIndex'